packages/ai-jsx/src/core/memoize.tsx

```diff
@@ -1,60 +1,28 @@
-import * as AI from '../index.js';
-import { RenderContext, Node, Renderable } from '../index.js';
+import { Renderable, RenderContext, AppendOnlyStream } from './render.js';
+import { Node, getReferencedNode, isIndirectNode, makeIndirectNode, isElement } from './node.js';
 import { Logger } from './log.js';
 
 let memoizedId = 0;
+/** @hidden */
 export const isMemoizedSymbol = Symbol('isMemoized');
 
 /**
- * Memoize a {@link Renderable} so it always returns the same thing.
- *
- * For example, imagine you have the following:
- * ```tsx
- *    const catName = (
- *      <ChatCompletion>
- *        <UserMessage>Give me a cat name</UserMessage>
- *      </ChatCompletion>
- *    );
- *
- *    <ChatCompletion>
- *      <UserMessage>
- *        I have a cat named {catName}. Tell me a story about {catName}.
- *      </UserMessage>
- *     </ChatCompletion>
- * ```
- *
- * In this case, `catName` will result in two separate model calls, so you'll get two different cat names.
- *
- * If this is not desired, you can wrap the component in `memo`:
- * ```tsx
- *    const catName = memo(<ChatCompletion>
- *      <UserMessage>Give me a cat name</UserMessage>
- *    </ChatCompletion>);
- *
- *    <ChatCompletion>
- *      <UserMessage>
- *        I have a cat named {catName}. Tell me a story about {catName}.
- *      </UserMessage>
- *     </ChatCompletion>
- * ```
- * Now, `catName` will result in a single model call, and its value will be reused everywhere that component appears
- * in the tree.
- *
- * The memoization is fully recursive.
+ * "Partially" memoizes a renderable such that it will only be rendered once in any
+ * single `RenderContext`.
```

Comment:
Hmm, why did we make this change to the docstring?

Reply:
Ah, I see it was moved elsewhere – I think we should put a `@hidden` on this function.